### PR TITLE
Replace Gitter with Discord

### LINKED ISSUE #886: DOC replace Gitter with Discord on the webpage
#### Describe the issue linked to the documentation

Our website points people to Gitter even though our Discord is much more active and not mentioned anywhere. We should replace occurrences of Gitter with Discord.

- nav bar at the top 
image find code at `docs/conf.py`, specifically [these lines](https://github.com/fairlearn/fairlearn/blob/378f1e693cff01c69de3a9d9504b336ae1e0064e/docs/conf.py#L134)
- [landing page](https://fairlearn.org/) 
image find code at `docs/static_landing_page/index.html`
- https://fairlearn.org/main/contributor_guide/index.html#real-time-communication, find code at `docs/contributor_guide/index.rst`

Note that older versions will still say Gitter so we can't delete our Gitter or abandon it completely, but we might as well point people in the right direction without adding another hop.

### PR DESCRIPTION
**Pull request by shimst3r**
closes #886 

I replace links to Gitter with links to Discord on the landing page and
within the documentation. I do not replace links in the README or the
GitHub templates.

### CODE REVIEW

File: docs/conf.py
Comment (hildeweerts) [inline docs/conf.py:136]: Could you also adjust the hover behavior of the `fa-discord` icon in the css? You can change that here: https://github.com/fairlearn/fairlearn/blob/main/docs/_static/css/custom.css

For this you can use the official discord color in this case (#5765f2).

**Review by hildeweerts — CHANGES_REQUESTED** [review]
Many thanks, @shimst3r!

I requested a few changes, let me know if anything is unclear!

File: docs/conf.py
Comment (shimst3r) [reply docs/conf.py:136]: Ah, the `custom.css` has slipped my search somehow. Thanks, it's resolved now. :+1:

File: docs/static_landing_page/index.html
Comment (shimst3r) [reply docs/static_landing_page/index.html:212]: I didn't lock the aspect ratio when I adjusted the image resolution. :sweat_smile: Should be fixed now!

**Review by hildeweerts — APPROVED** [review]
Perfect!

@fairlearn/fairlearn-maintainers feel free to merge if you don't have any further comments

**Comment by adrinjalali** [comment]
We don't wanna add this to the page people see when they wanna open an issue?

Screenshot_20210709-165222

**Comment by hildeweerts** [comment]
@adrinjalali I'm also happy to leave that to a separate PR, but @shimst3r you can change the template here: https://github.com/fairlearn/fairlearn/blob/main/.github/ISSUE_TEMPLATE/config.yml

**Comment by hildeweerts** [comment]
Same goes for the README.md btw @shimst3r

**Comment by shimst3r** [comment]
@adrinjalali I had the same idea, but didn't know whether @romanlutz had any reasons for not including it in #886. 😄 

@hildeweerts in order to replace the Gitter shield with the Discord shield, the necessary feature needs to be activated on our Discord server. Do you know how to do this?

**Comment by romanlutz** [comment]
@shimst3r thanks for being more thorough than I was in creating the issue 🙂 
What do you mean by enabling the shield?

**Comment by romanlutz** [comment]
Thanks for this awesome contribution @shimst3r ! This ended up being a lot more than I had thought with the landing page 😮

**Pull request merged**
